### PR TITLE
QED on-demand backup

### PR DESCRIPTION
**Pull request by jllucas**
2 functionalities:
- create backups
- lists backups with info

### CODE REVIEW

File: cmd/backup.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+	"github.com/octago/sflags/gen/gpflag"
+	"github.com/spf13/cobra"
+
+	"github.com/bbva/qed/log"
+)
+
+type BackupConfig struct {
+	// Endpoint [host:port] to ask for QED management APIs.
+	Endpoint string `desc:"Management QED Log service endpoint http://ip:port"`
```
Comment (aalda) [inline cmd/backup.go]: Replace description with "QED Log service management endpoint...

File: rocksdb/backup.go [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+// SetKeepLogFiles is used to set or unset the keep_log_files option
+// If true, restore won't overwrite the existing log files in wal_dir. It will
+// also move all log files from archive directory to wal_dir.
+// By default, this is false.
+func (ro *RestoreOptions) SetKeepLogFiles(v int) {
+	C.rocksdb_restore_options_set_keep_log_files(ro.c, C.int(v))
+}
+
+// Destroy destroys this RestoreOptions instance.
+func (ro *RestoreOptions) Destroy() {
```
Comment (aalda) [inline rocksdb/backup.go]: Use Destroy()

File: rocksdb/backup.go [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+}
+
+// GetAppMetadata gets the backup associated metadata.
+func (b *BackupEngineInfo) GetAppMetadata(index int) string {
+	metadata := C.rocksdb_backup_engine_info_metadata(b.c, C.int(index))
+	return C.GoString(metadata)
+}
+
+// Destroy destroys the backup engine info instance.
+func (b *BackupEngineInfo) Destroy() {
```
Comment (aalda) [inline rocksdb/backup.go]: Use Destroy()

File: raftwal/fsm.go [hunk 4]
```diff
@@ -29,6 +29,7 @@ import (
 	"github.com/bbva/qed/log"
 	"github.com/bbva/qed/raftwal/commands"
 	"github.com/bbva/qed/storage"
+
 	"github.com/hashicorp/go-msgpack/codec"
```
Comment (iknite) [inline raftwal/fsm.go]: hashicorp should go before our lib

File: raftwal/fsm.go [hunk 5]
```diff
@@ -273,6 +274,27 @@ func (fsm *BalloonFSM) Restore(rc io.ReadCloser) error {
 	return fsm.balloon.RefreshVersion()
 }
 
+// Backup ...
+func (fsm *BalloonFSM) Backup() error {
+	fsm.restoreMu.Lock()
```
Comment (iknite) [inline raftwal/fsm.go]: This blocks all the fsm or only backup stack?

**Pull request merged**